Automattic/woocommerce-payments
# Dev/paginated request class

Fixes #5058 

#### Add Paginated request classes

New paginated request class have been created. Using this class, you can easily execute paginated API call to wc pay server.

Here is the example of paginated request:

```
 $page      = 2;
$page_size = 50;
$direction = 'asc';
$sort      = 'created';
$filters   = [ 'key'  => 'value' ];
$uri       = 'uri';

$request = Paginated::create();
$request->set_page( $page );
$request->set_page_size( $page_size );
$request->set_sort_direction( $direction );
$request->set_sort_by( $sort );
$request->set_filters( $filters );
$request->set_uri( $uri );
```
This class will return the same response from wc pay server. For simple requests where you need results from the server without formatting or modeling response data, use this paginated request. Since there could be plenty of similar requests where URI or URL is the only difference, the `set_uri` method provide an easy way to define API URI.

There are `List_Disputes` and `List_Transactions` request classes that are similar like paginated request class, but the only difference is that these classes format responses.









#### Testing instructions





* Make sure that all tests are passing
* Test new request classes on your environment and make sure that everything is working as expected.

-------------------


**Post merge**
